src/util/decode-bitcoin-address.ts

```diff
@@ -0,0 +1,26 @@
+import { AddressType, getAddressInfo } from "bitcoin-address-validation";
+import * as bitcoin from "bitcoinjs-lib";
+
+const typeMapping = {
+  [AddressType.p2pkh]: "0x00",
+  [AddressType.p2sh]: "0x01",
+  [AddressType.p2wpkh]: "0x04",
+  [AddressType.p2wsh]: "0x05",
+  [AddressType.p2tr]: "0x06",
+};
+export function decodeBitcoinAddress(address: string) {
+  const addressInfo = getAddressInfo(address);
+
+  const { bech32 } = addressInfo;
+  let hash: Uint8Array;
+  if (bech32) {
+    hash = bitcoin.address.fromBech32(address).data;
+  } else {
+    hash = bitcoin.address.fromBase58Check(address).hash;
+  }
+
+  return {
+    type: typeMapping[addressInfo.type],
```

Comment:
I'm wondering if we should ensure `addressInfo.type` exists in `typeMapping`: right now that is always the case, but if `bitcoin-address-validation` were to update and introduce some other `addresstype`, this would return undefined and maybe bad things could happen downstream

Reply:
that is covered since the package is pinned to the exact version that supports all these types atm and will always do since the version is pinned

Reply:
It is, right now, but if we update the dependency there's nothing that will flag if there's a new address type we don't support.

Reply:
Fair enough will throw an error if the returned type is incorrect